Emrald-UI/src/contexts/LogicNodeContext.tsx

```diff
@@ -55,10 +55,14 @@ const LogicNodeContextProvider: React.FC<EmraldContextWrapperProps> = ({
   const logicNodeList = useComputed(() => appData.value.LogicNodeList);
 
 
-  const createLogicNode = (newLogicNode: LogicNode) => {
-    const updatedLogicNodes = [...appData.peek().LogicNodeList, newLogicNode];
-    updateAppData(JSON.parse(JSON.stringify({ ...appData.value, LogicNodeList: updatedLogicNodes })));
-    setLogicNodes(updatedLogicNodes);
+
+  const createLogicNode = async (newLogicNode: LogicNode) => {
```

Comment:
Did you verify on these that they were all getting added correctly for all of them?

Reply:
Yes, they are all verified except for the ExtSimContext one